specs/commandline/plugin.md

```diff
@@ -41,51 +42,160 @@ Aliases:
 ### notation plugin install
 
 ```text
-Installs a plugin
+Install a plugin
 
 Usage:
-  notation plugin install [flags] <plugin path>
+  notation plugin install [flags] <plugin_source>
 
 Flags:
   -h, --help                    help for install
   -f, --force                   force the installation of a plugin
+      --checksum string         must match SHA256 of the plugin source               
 
 Aliases:
   install, add
 ```
 
-### notation plugin remove
+### notation plugin upgrade
 
 ```text
-Removes a plugin
+Upgrade a plugin
```

Comment:
I also expect allow users to upgrade without specifying the URL. It seems users still need to manually get the sha256sum from the plugin download page

Reply:
> How does the proposed behavior differ from using the install command?

I expect upgrade a plugin without specifying the plugin URL. This differs from the `notation plugin install` experience. I will update the CLI spec to describe this experience. 

> Could the install command be used to handle the following scenarios:

For 1, it will be handled by current design.
For 2 and 3,  the logic is a little different. `notation plugin install` only checks the plugin existence instead of version.  
the current design and behavior are described at [this line](https://github.com/notaryproject/notation/pull/809/files#diff-7637fbf27d1c8c25ef1a9806613d4aa17aaa61d99aae29f8c802b5e50bcd2efdR105). In short, when an existing plugin is detected, it fails to install and returns the error. Users can use a flag `--force` to skip existence check and force the installation.

Reply:
@priteshbandi I have adopted your suggestions above and update the `notation plugin install` usage and examples accordingly. Thanks